core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -1897,6 +1930,29 @@ public static double atan2(double b0, double b1) {
     return Math.atan2(b0, b1);
   }
 
+  // ATANH
+  /** SQL <code>ATANH</code> operator applied to BigDecimal values. */
+  public static double atanh(BigDecimal b) {
+    return atanh(b.doubleValue());
+  }
+
+  /** SQL <code>ATANH</code> operator applied to double values. */
+  public static double atanh(double b) {
+    if (b > 1 || b < -1) {
```

Comment:
According to [this](https://en.wikipedia.org/wiki/Inverse_hyperbolic_functions) and [this](https://support.microsoft.com/en-us/office/atanh-function-3cd65768-0de7-4f1d-b312-d01c8c930d90), I think that 1 and -1 should also be forbidden:
```
if (b => 1 || b <= -1) {
  throw new IllegalArgumentException(...);
}
```